spring-ai-model/src/main/java/org/springframework/ai/chat/prompt/PromptTemplate.java

```diff
@@ -179,59 +190,89 @@ public Prompt create(ChatOptions modelOptions) {
 	}
 
 	@Override
-	public Prompt create(Map<String, Object> model) {
-		return new Prompt(render(model));
+	public Prompt create(Map<String, Object> additionalVariables) {
+		return new Prompt(render(additionalVariables));
 	}
 
 	@Override
-	public Prompt create(Map<String, Object> model, ChatOptions modelOptions) {
-		return new Prompt(render(model), modelOptions);
+	public Prompt create(Map<String, Object> additionalVariables, ChatOptions modelOptions) {
+		return new Prompt(render(additionalVariables), modelOptions);
 	}
 
+	// Compatibility
+
+	/**
+	 * @deprecated in favor of {@link TemplateRenderer}.
+	 */
+	@Deprecated
 	public Set<String> getInputVariables() {
-		TokenStream tokens = this.st.impl.tokens;
-		Set<String> inputVariables = new HashSet<>();
-		boolean isInsideList = false;
-
-		for (int i = 0; i < tokens.size(); i++) {
-			Token token = tokens.get(i);
-
-			if (token.getType() == STLexer.LDELIM && i + 1 < tokens.size()
-					&& tokens.get(i + 1).getType() == STLexer.ID) {
-				if (i + 2 < tokens.size() && tokens.get(i + 2).getType() == STLexer.COLON) {
-					inputVariables.add(tokens.get(i + 1).getText());
-					isInsideList = true;
-				}
-			}
-			else if (token.getType() == STLexer.RDELIM) {
-				isInsideList = false;
-			}
-			else if (!isInsideList && token.getType() == STLexer.ID) {
-				inputVariables.add(token.getText());
-			}
-		}
+		throw new UnsupportedOperationException(
```

Comment:
This is the only behavioural breaking change in this PR. What do you think?

This method was used internally for validation and testing (and it's still used that way, but inside StTemplateRenderer). This is very specific not only based on the template engine, but also on how it's configured, so I wouldn't expose this information here. If we need some additional introspection into the internals of the template engine, it should be the TemplateRenderer giving access to that.